streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -685,13 +685,13 @@ public boolean test(final K1 key, final V1 value) {
                                                                                     windows.maintainMs());
 
             KStreamKStreamJoin<K1, R, V1, V2> joinThis = new KStreamKStreamJoin<>(otherWindow.name(),
-                                                                                  windows.before,
                                                                                   windows.after,
+                                                                                  windows.before,
```

Comment:
I think the documentation you are referring to is incorrect. My understanding is this:
`stream1.join(stream2, JoinWindows.of(0).before(100))` means that records from `stream2` will be joined with records from `stream1` if `stream2.ts`  is >= `stream1.ts - 100` and `stream2.ts` is <= `stream1.ts + 0`

So given: 
`stream1.ts = 1000` and `stream2.ts = 900` then `900 >= 900 && 900 <= 1000` -> true
`stream1.ts = 1000` and `stream2.ts=1000` then `1000 >= 1000 && 1000 <= 1000` -> true
`stream1.ts = 1000` and `stream2.ts=1100` then `1100 >= 1000 && 1100 <= 1000` -> false